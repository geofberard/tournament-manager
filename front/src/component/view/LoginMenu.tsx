import { Grid, SxProps, Theme, Typography } from "@mui/material";
import * as React from "react";
import { FC } from "react";
import { Image } from "../atom/Image";

const menuContainer: SxProps = {
    backgroundColor: "primary.dark",
    color: "primary.contrastText",
};

const logoContainer: SxProps = {
    display: "flex",
    alignItems: "center",
};

const titleContainer: SxProps = {
    ...logoContainer,
    paddingBottom: { xs: "0", md: "30px" },
    borderBottom: (theme: Theme) => `5px solid ${theme.palette.secondary.main}`,
};

const logo: SxProps = {
    width: { xs: "75px", md: "250px" },
};

export const LoginMenu: FC = () => (
    <Grid
        container
        item
        md={5}
        direction={{ xs: "row", md: "column-reverse" }}
        justifyContent={{ xs: "space-between", md: "center" }}
        alignItems={{ xs: "stretch", md: "center" }}
        sx={menuContainer}
    >
        <Grid item sx={logoContainer}>
            <Image src="/img/scuf-logo.svg" alt="SCUF" sx={logo} />
        </Grid>
        <Grid item sx={titleContainer}>
            <Typography variant="h1">Tournois</Typography>
        </Grid>
        <Grid item />
    </Grid>
);
