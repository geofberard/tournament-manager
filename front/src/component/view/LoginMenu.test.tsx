import * as React from 'react';
import { testSnapshot } from '../../test/TestUtils';
import { LoginMenu } from './LoginMenu';

testSnapshot(<LoginMenu />)