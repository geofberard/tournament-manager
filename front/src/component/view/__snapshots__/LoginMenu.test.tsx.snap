// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match last snapshot 1`] = `
<ForwardRef(Grid)
  alignItems={
    Object {
      "md": "center",
      "xs": "stretch",
    }
  }
  container={true}
  direction={
    Object {
      "md": "column-reverse",
      "xs": "row",
    }
  }
  item={true}
  justifyContent={
    Object {
      "md": "center",
      "xs": "space-between",
    }
  }
  md={5}
  sx={
    Object {
      "backgroundColor": "primary.dark",
      "color": "primary.contrastText",
    }
  }
>
  <ForwardRef(Grid)
    item={true}
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <Image
      alt="SCUF"
      src="img/scuf-logo.svg"
      sx={
        Object {
          "width": Object {
            "md": "250px",
            "xs": "75px",
          },
        }
      }
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    sx={
      Object {
        "alignItems": "center",
        "borderBottom": [Function],
        "display": "flex",
        "paddingBottom": Object {
          "md": "30px",
          "xs": "0",
        },
      }
    }
  >
    <ForwardRef(Typography)
      variant="h1"
    >
      Tournois
    </ForwardRef(Typography)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
  />
</ForwardRef(Grid)>
`;
